tests/sessions/claims/test_verify_session.py

```diff
@@ -0,0 +1,453 @@
+    res = fastapi_client.post("create-with-claim", json={"key": "something"})
+    return res
+
+
+async def test_should_allow_without_claims_required_or_present(
+    fastapi_client: TestClient,
+):
+    st_init_args = {
+        **st_init_common_args,
+        "recipe_list": [session.init(anti_csrf="VIA_TOKEN")],
+    }
+    init(**st_init_args)  # type: ignore
+    start_st()
+
+    create_session(fastapi_client)
+
+    res = fastapi_client.get("/default-claims")
+    assert res.status_code == 200
+    assert "-" in res.json()["handle"]
+
+
+async def test_should_allow_with_claim_valid_after_refetching(
+    fastapi_client: TestClient,
+):
+    st_init_args = st_init_generator_with_claim_validator(
+        TrueClaim.validators.has_value(True)
+    )
+    init(**st_init_args)  # type: ignore
+    start_st()
+
+    create_session(fastapi_client)
+
+    response = fastapi_client.get("/default-claims")
+    assert response.status_code == 200
+
+
+async def test_should_reject_with_claim_required_but_not_added(
+    fastapi_client: TestClient,
+):
+    st_init_args = st_init_generator_with_claim_validator(
+        NoneClaim.validators.has_value(True)  # type: ignore
+    )
+    init(**st_init_args)  # type: ignore
+    start_st()
+
+    create_session(fastapi_client)
+
+    response = fastapi_client.get("/default-claims")
+    assert response.status_code == 403
+    assert response.json() == {
+        "message": "invalid claim",
+        "claimValidationErrors": [
+            {
+                "id": "st-none",
+                "reason": {
+                    "message": "wrong value",
+                    "expectedValue": True,
+                    "actualValue": None,  # TODO: Do we want to return actualValue or not?
+                },
+            }
+        ],
+    }
+
+
+async def test_should_allow_with_custom_validator_returning_true(
+    fastapi_client: TestClient,
+):
+    custom_validator = AlwaysValidValidator()
+
+    st_init_args = st_init_generator_with_claim_validator(custom_validator)
+    init(**st_init_args)  # type: ignore
+    start_st()
+
+    create_session(fastapi_client)
+    res = fastapi_client.get("/default-claims")
+    assert res.status_code == 200
+    assert "-" in res.json()["handle"]
+
+
+async def test_should_reject_with_custom_validator_returning_false(
+    fastapi_client: TestClient,
+):
+    custom_validator = AlwaysInvalidValidator()
+
+    st_init_args = st_init_generator_with_claim_validator(custom_validator)
+    init(**st_init_args)  # type: ignore
+    start_st()
+
+    create_session(fastapi_client)
+    response = fastapi_client.get("/default-claims")
+    assert response.status_code == 403
+    assert response.json() == {
+        "message": "invalid claim",
+        "claimValidationErrors": [{"id": "always-invalid-validator", "reason": "foo"}],
+    }
+
+
+# should reject with validator returning false with reason (Leaving this. It's exactly same as prev.)
```

Comment:
I'll check/update this in the node PR, but one should return it with the reason set to something, one without.

Reply:
Okay. TODO

Reply:
@porcellus @KShivendu what is the status of this comment?